src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java

```diff
@@ -44,28 +47,52 @@ public final void publishWithDelay(Message message, long delayMilliseconds) thro
             log.warn("Publishing delayed message to non-delayed queue queue:{}", queueName);
         }
 
+        val properties = new AMQP.BasicProperties.Builder();
+
+        if (config.getCompressionConfig() != null &&
+                config.getCompressionConfig().isEnableCompression() &&
+                config.getCompressionConfig().getCompressionAlgorithm() != null) {
+            properties.headers(ImmutableMap.of("X-CompressionType",
```

Comment:
`X-CompressionType` -> `x-compression-type`

Reply:
Simlar to handler, it is better to introduce a decorator here .. add headers, modified message etc and use that top fire the `publish()` in one shot